src/benchmarks/gc/src/exec/GCPerfSim/GCPerfSim.cs

```diff
@@ -769,29 +772,180 @@ public override string ToString()
+        {
+            if (sizeSlots[i].size < limitSizeRange.low || limitSizeRange.high < sizeSlots[i].size)
+                continue;
+            SizeRange sizeRange = new SizeRange(lowSize + overhead + 1, sizeSlots[i].size + overhead + 1);
+            lowSize = sizeSlots[i].size;
+            double weight = (double)sizeSlots[i].count * Math.Max(survInterval, 1);
+            BucketSpec bucketSpec = new BucketSpec(sizeRange, survInterval, reqSurvInterval, pinInterval, finalizableInterval, weight, isPoh);
+            bucketSpecs.Add(bucketSpec);
+        }
+    }
+    public static void BuildSOHBucketSpecsFromSizeDistribution(List<BucketSpec> bucketSpecs, uint survInterval, uint reqSurvInterval, uint pinInterval, uint finalizableInterval)
+    {
+        BuildBucketSpecsFromSizeDistribution(bucketSpecs, sohLohSizeSlots, new SizeRange(48, 84_999), survInterval, reqSurvInterval, pinInterval, finalizableInterval, false);
+    }
+    public static void BuildLOHBucketSpecsFromSizeDistribution(List<BucketSpec> bucketSpecs, uint survInterval, uint reqSurvInterval, uint pinInterval, uint finalizableInterval)
+    {
+            BuildBucketSpecsFromSizeDistribution(bucketSpecs, sohLohSizeSlots, new SizeRange(85_000, uint.MaxValue), survInterval, reqSurvInterval, pinInterval, finalizableInterval, false);
+    }
+
+    // an object size distribution for OH derived from a real server scenario
+    public static SizeSlot[] pohSizeSlots = new SizeSlot[]
+    {
+        new SizeSlot(        56,        16),
+        new SizeSlot(      1048,         1),
+        new SizeSlot(      2072,         1),
+        new SizeSlot(      4120,         1),
+        new SizeSlot(      8184,         2),
+        new SizeSlot(      8216,         1),
+        new SizeSlot(     16344,         2),
+        new SizeSlot(     16408,         1),
+        new SizeSlot(     32664,         2),
+        new SizeSlot(     32792,         1),
+        new SizeSlot(     65304,         2),
+        new SizeSlot(     65560,         1),
+        new SizeSlot(    130584,         1),
+        new SizeSlot(    131064,       160),
+        new SizeSlot(   2259408,         1),
+    };
+
+    public static void BuildPOHBucketSpecsFromSizeDistribution(List<BucketSpec> bucketSpecs, SizeSlot[] sizeSlots, SizeRange sizeRange, uint survInterval, uint reqSurvInterval, uint pinInterval, uint finalizableInterval)
+    {
+        BuildBucketSpecsFromSizeDistribution(bucketSpecs, sizeSlots, sizeRange, survInterval, reqSurvInterval, pinInterval, finalizableInterval, true);
+    }
+    public static void BuildPOHBucketSpecsFromSizeDistribution(List<BucketSpec> bucketSpecs, uint survInterval, uint reqSurvInterval, uint pinInterval, uint finalizableInterval)
+    {
+        BuildBucketSpecsFromSizeDistribution(bucketSpecs, pohSizeSlots, new SizeRange(24, 10_000_000), survInterval, reqSurvInterval, pinInterval, finalizableInterval, true);
+    }
+}
+
+
 readonly struct Phase
```

Comment:
it'd be great to have a sample phase txt checked in

Reply:
Done.